client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -0,0 +1,330 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import ReactDomServer from 'react-dom/server';
+import { connect } from 'react-redux';
+import { includes } from 'lodash';
+import tinymce from 'tinymce/tinymce';
+
+/**
+ * Internal dependencies
+ */
+import SuggestionList from './suggestion-list';
+import EditorMention from './editor-mention';
+import QueryUsersSuggestions from 'components/data/query-users-suggestions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getUserSuggestions } from 'state/users/suggestions/selectors';
+
+/**
+ * Module variables
+ */
+const VK = tinymce.util.VK;
+
+const getMatchingSuggestions = function( suggestions, query ) {
+	if ( ! query ) {
+		return suggestions.slice( 0, 10 );
+	}
+
+	return suggestions
+		.filter( ( { user_login, display_name } ) => {
+			// Start of string or preceded by a space.
+			const matcher = new RegExp( `^${ query }|\\s${ query }`, 'ig' );
```

Comment:
Re: whitespace. It's matching against `${ user_login } ${ display_name }`, so whether the regex uses `\s` or `\s+`, the result would be the same.

I also tried creating a user with a user name starting with ^ and it wasn't allowed.